component/firecracker-base/prepare_jailer.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+set -euo pipefail
+
+SB_ID="${1:-0}" # Default to sb_id=0
+
+DATA_DIR="/firecracker-data"
+JAILER_DIR="/srv/jailer/firecracker"
+JAILER_BINARY="/usr/bin/jailer"
+
+ROOTFS="rootfs.ext4"
+KERNEL="image-kernel.bin"
+
+RO_DRIVE="$DATA_DIR/$ROOTFS"
+KERNEL_IMG="$DATA_DIR/$KERNEL"
+
+TAP_DEV="fc-${SB_ID}-tap0"
+FC_MAC="$(printf '02:FC:00:00:%02X:%02X' $((SB_ID / 256)) $((SB_ID % 256)))"
+VETH_DEV="veth-jailer$SB_ID"
+JAILER_NS="jailer-$SB_ID"
+
+########## ############################# #########
+##########           User Prep           #########
+########## ############################# #########
+
+# Create a user and group to run the execution via for one micro-vm
+if id 10000$SB_ID >/dev/null 2>&1; then
+  echo "User 10000$SB_ID already exists, skipping creation"
+else
+  echo "Creating user 10000$SB_ID"
+  useradd -M -u 10000$SB_ID $JAILER_NS
+  usermod -L $JAILER_NS
+
+  # This was created earlier on the machine provisioning
+  usermod -a -G jailer-processes $JAILER_NS
+  usermod -a -G root $JAILER_NS
+  usermod -a -G kvm $JAILER_NS
+fi
+
+########## ############################# #########
+##########          Disk Prep            #########
+########## ############################# #########
+
+JAIL="$JAILER_DIR/$SB_ID/root"
+mkdir -p "$JAIL/"
+
+touch $JAIL/logs
+touch $JAIL/metrics
+
+if test -f "$JAIL/$KERNEL"; then
+  echo "Jailed kernel exists, skipping creation."
+else
+  echo "Copying kernel image..."
+  cp -v $KERNEL_IMG "$JAIL/$KERNEL"
+fi
+
+if test -f "$JAIL/$ROOTFS"; then
+  echo "Jailed rootfs exists, skipping creation."
+else
+  echo "Copying rootfs..."
+  cp -v $RO_DRIVE "$JAIL/$ROOTFS"
+  # using device mapper for CoW should be faster. Something like this?
```

Comment:
query: If this works I'd totally suggest we include it, the initial implementation of `cp` doesn't scale at all.

Reply:
I started to mess with it, but it was tricky. I'd like to push it out to another PR so I can focus on it specifically

Reply:
Sounds good 👍